### PR TITLE
First draft of SIP feature

### PR DESCRIPTION
**Pull request by digitaltsai**
This is a first draft of a possible SIP feature for OpenTok. This would allow the PHP OpenTok SDK to dial a SIP Uri and have them join the session.

### CODE REVIEW

**Comment by digitaltsai** [comment]
@aoberoi please review when you get the chance. I'm not sure if I should have all the credentials passed as part of the options array, or as separate parameters.

File: src/OpenTok/OpenTok.php [hunk 1]
```diff
@@ -385,6 +385,74 @@ public function listArchives($offset=0, $count=null)
...
+     *    custom headers must start with the "X-" prefix, or a Bad Request (400) will be thrown.</li>
+     *
+     *    <li><code>'secure'</code> (int) &mdash; Secure​: Boolean (trueor false) flag that indicates
+     *    whether the media must be transmitted encrypted or not.</li>
+     *
+     * </ul>
+     *
+     * @return SipCall The SipCall, which contains the ids of the Sip connection.
+     */
+    public function call($sessionId, $token, $sipUri, $username, $password, $options=array())
```
Comment (aoberoi) [inline src/OpenTok/OpenTok.php]: i think we're going with `dial()` as the name of the API. we have a few weeks to really decide, but i just wanted to comment on where most of the consensus is right now.

Comment (aoberoi) [inline src/OpenTok/OpenTok.php]: Can we make `$username` and `$password` part of the options?

**Comment by aoberoi** [comment]
it would be nice to have more than a single test. some ideas:
*  using optional parameters and/or not using them
*  getting a bad response (500)

**Comment by aoberoi** [comment]
FYI: I've asked that all new API go through an open API Review. This is done by creating a GitHub Issue, with the label `question` and requesting others to comment. That link should also be shared internally in #api-review. Here is an example: https://github.com/opentok/OpenTok-PHP-SDK/issues/156

File: src/OpenTok/OpenTok.php (hunk 1 above)
Comment (digitaltsai) [inline src/OpenTok/OpenTok.php]: Are username and password optional?

Comment (ggarber) [inline src/OpenTok/OpenTok.php]: yes, they are optional

File: sample/SipCall/web/index.php [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+use OpenTok\MediaMode;
+
+// PHP CLI webserver compatibility, serving static files
+$filename = __DIR__.preg_replace('#(\?.*)$#', '', $_SERVER['REQUEST_URI']);
+if (php_sapi_name() === 'cli-server' && is_file($filename)) {
+    return false;
+}
+
+// Verify that the API Key and API Secret are defined
+if (!(getenv('API_KEY') && getenv('API_SECRET'))) {
```
Comment (aoberoi) [inline sample/SipCall/web/index.php]: what about other environment variables

File: sample/SipCall/web/index.php [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+$app->container->singleton('cache', function() {
+    return new Cache;
+});
+
+// Initialize OpenTok instance, store it in the app contianer
+$app->container->singleton('opentok', function () {
+    return new OpenTok(getenv('API_KEY'), getenv('API_SECRET'));
+});
+// Store the API Key in the app container
+$app->apiKey = getenv('API_KEY');
```
Comment (aoberoi) [inline sample/SipCall/web/index.php]: getenv's go here

File: sample/SipCall/web/js/opentok-layout.min.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+/*!
```
Comment (aoberoi) [inline sample/SipCall/web/js/opentok-layout.min.js]: @digitaltsai it looks like this file isn't used, can we remove it?

File: src/OpenTok/OpenTok.php [hunk 5]
```diff
@@ -385,6 +385,87 @@ public function listArchives($offset=0, $count=null)
...
+    /**
+     * Initiate an outgoing SIP call
+     *
+     * @param string $sessionId The OpenTok SessionIdwhere the participant being called
+     * will join.
+     *
+     * @param string $token The token for conecting to an OpenTok session. This is the same
+     * as the one created by Session.generateToken.
+     *
+     * @param string $username Auth​: Username to be used in the SIP INVITE request for HTTP
```
Comment (aoberoi) [inline src/OpenTok/OpenTok.php]: these docs are outdated, the username and password are part of options now. @jeffswartz will have to do a final pass anyway so i'm fine with merging this now.

**Comment by aoberoi** [comment]
:tada: :calling:

**Pull request merged**